### PR TITLE
MASF

### PR DESCRIPTION
**Pull request by jodoin**
A domain adaptation method.

### CODE REVIEW

File: article/_posts/2020-11-18-MASF.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    venue:   "NeurIPS 2019"
+pdf: "https://arxiv.org/pdf/1910.13580.pdf"
+---
+
+
+![](/article/images/MASF/sc01.jpg)
+
+
+# Highlights
+This paper propose a new domain generalization method.  The goal is for a model, trained on multi-domain source data, to generalize well on target domains with unknown statistics.  The proposed system is illustred in Figigure 1.
```
Comment (jhlegarreta) [inline article/_posts/2020-11-18-MASF.md:18]: propose : proposes
Figigure 1 : Figure 1

File: article/_posts/2020-11-18-MASF.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+Say we have various domains $$D=\{D_1,D_2,...,D_K\}$$ and in each domain we have a series of labeled data $$D_k=\{(x_n^k,y_n^k)\}_{n=1}^{N_k}$$.  The proposed system comes with a feature extractor $$F_\phi$$ and a task network $$T_\theta$$.  The predicted output for a given input $$x$$ is $$\hat{y}=softmax(T_\theta(F_\phi(x)))$$. 
+
+At each iteration, $$D$$ is split into **meta-train** $$D_{tr}$$ and **meta-test** $$D_{te}$$ domains. Then, as shown in Figure 1, 3 losses are being optimised.
+
+* A *task* loss called $${\cal L}_{task}$$  (a basic cross-entropy). 
+* A *global* loss $${\cal L}_{global}$$
+* A *local* loss $${\cal L}_{local}$$
+
+The goal of the *global* loss  is to make sure that the class relationship is the same across domain. This is done by computing the average latent vector for every data associated to each class $$c$$ and each domain $$k$$
```
Comment (jhlegarreta) [inline article/_posts/2020-11-18-MASF.md:30]: domain : domains

File: article/_posts/2020-11-18-MASF.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+![](/article/images/MASF/sc02.jpg)
+
+Then, take the average latent vector, decode it and compute it *soft*-crossentropy
```
Comment (jhlegarreta) [inline article/_posts/2020-11-18-MASF.md:34]: compute it *soft*-crossentropy: compute its *soft*-cross-entropy

File: article/_posts/2020-11-18-MASF.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+![](/article/images/MASF/sc03.jpg)
+
+where $$\tau$$ is a temperature term.  The global loss is the following Jensen-Shannon divergence
+
+![](/article/images/MASF/sc04.jpg)
+
+
+As for the *local* loss, its goal is to enforce that a latent vector is closer to another latent vectir from the same class than one from from other classes.  For this, they propose a **contractive loss** or a **triplet loss**
```
Comment (jhlegarreta) [inline article/_posts/2020-11-18-MASF.md:43]: vectir: vector

Comment (jhlegarreta) [inline article/_posts/2020-11-18-MASF.md:43]: Not sure if "contractive" and "contrastive" are used interchangeably in the literature.

File: article/_posts/2020-11-18-MASF.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+![](/article/images/MASF/sc05.jpg)
```
Comment (jhlegarreta) [inline article/_posts/2020-11-18-MASF.md:45]: Maybe add a blank line so that both images are displayed on different "lines"?

File: article/_posts/2020-11-18-MASF.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+The overall training algorithm goes as follows
+
+![](/article/images/MASF/sc07.jpg)
+
+
+
+# Results
+
+
+The authors report good recognition accuracies on the PACS dataset (a dataset with challenging domain shift).  Notice that **DeepAll** is their baseline for which all domains are merged and $$F_\phi.T_\theta$$ is trained by standard supervised learning with $${\cal L}_{task}$$.  Interestingly, DeepAll beats several SOTA methods!
```
Comment (jhlegarreta) [inline article/_posts/2020-11-18-MASF.md:58]: $$F_\phi$$ and $$T_\theta$$ may be more clear?

File: article/_posts/2020-11-18-MASF.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+
+![](/article/images/MASF/sc08.jpg)
+![](/article/images/MASF/sc09.jpg)
+
+
+Unfortunately, their method does not seem to work on medical images
```
Comment (jhlegarreta) [inline article/_posts/2020-11-18-MASF.md:65]: Not sure if I get this; probably not as good as networks testing on the same training domain, but is a 0.9 Dice clearly not enough?

**Review by jhlegarreta — COMMENTED** [review]
Similar to DIVA in its quest for a good domain generalization, but different approach. Would have been interesting to see a comparison; DIVA seems more recent but this work is not referenced.

Some in-line comments.

File: article/_posts/2020-11-18-MASF.md (hunk 2 above)
Comment (nathanpainchaud) [reply article/_posts/2020-11-18-MASF.md:30]: I think singular _domain_ here is correct, but I'd reword the sentence to rather be:
> This is done by computing the average latent vectors, for each class $$c$$ and domain $$k$$, over their respective associated data.

Comment (jhlegarreta) [reply article/_posts/2020-11-18-MASF.md:30]: Sorry I meant the first appearance of "domain".

File: article/_posts/2020-11-18-MASF.md (hunk 6 above)
Comment (jodoin) [reply article/_posts/2020-11-18-MASF.md:58]: no, that is the correct notation.

File: article/_posts/2020-11-18-MASF.md (hunk 7 above)
Comment (jodoin) [reply article/_posts/2020-11-18-MASF.md:65]: their method is not better than naive "deepAll"

**Review by jhlegarreta — APPROVED** [review]
